app/src/main/java/com/spascoding/taskycourse/feature_register_screen/data/repository/AuthRepositoryImpl.kt

```diff
@@ -45,11 +46,15 @@ class AuthRepositoryImpl @Inject constructor(
         return response
     }
 
-    override fun authenticate() {
-        TODO("Not yet implemented")
+    override suspend fun authenticate(
+        token: String
+    ): Boolean {
+        val response = authenticationApi.authenticate(token)
+        return response.isSuccessful && response.code() == 200
```

Comment:
`response.isSuccessful` will suffice here. A response is successful if its status code starts with a 2. Your code will only work for exactly 200